py17track/client.py

```diff
@@ -35,9 +35,9 @@ async def _request(
         """Make a request against the RainMachine device."""
         use_running_session = self._session and not self._session.closed
 
-        if use_running_session:
+        if self._session is not None:
```

Comment:
Why? The previous logic checked this (and more, like whether the session was closed).

Reply:
mypy pre-commmit didn't let me commit due to typings conflicts. The final outcome was not changed and the next if checks if it is closed. Type assertion is nescery for the pre commit check to work

Reply:
The mypy pre commit hook first started screaming at me due to line 20, and after I fixed it, it started yelling at me about these lines...

Reply:
This can be more simply handled by inputting an assertion into the original code:

```python
use_running_session = self._session and not self._session.closed

if use_running_session:
    session = self._session
else:
    session = ClientSession(timeout=ClientTimeout(total=DEFAULT_TIMEOUT))

assert session
```

Reply:
@bachya just tried that and another gate is falling:

```
bandit...................................................................Failed
- hook id: bandit
- exit code: 1

py17track/client.py:43: B101[bandit]: LOW: Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.
```

Any other suggestions?

Reply:
My fault – just pushed, which should help.

Reply:
@bachya nice thanks, I just pushed the suggested fix